### ISSUE TITLE
Plugin installation / execution fails

### ISSUE THREAD

**Issue by flschr** [nasendackel/yellow-comments#26]
Hi Giovanni!
I added your plugin to my dev stage and it seems that i't was'nt installed correctly, as I can't find the files in my extension directory after putting the zip file there. So I added the files manually and also added the parameters to the yellow-system.ini. But when I access a blog post, I always see the following error (debugging enabled). How can I solve this?

`Fatal error: Uncaught Error: Call to undefined method YellowToolbox::getTextArgs() in /usr/www/users/gaehna/flow/new2/system/extensions/comments.php:61 Stack trace: #0 /usr/www/users/gaehna/flow/new2/system/extensions/comments.php(142): YellowComments->onParseContentShortcut(Object(YellowPage), 'comments', '', true) #1 /usr/www/users/gaehna/flow/new2/system/extensions/core.php(855): YellowComments->onParsePageExtra(Object(YellowPage), 'comments') #2 /usr/www/users/gaehna/flow/new2/system/layouts/blog.html(18): YellowPage->getExtra('comments') #3 /usr/www/users/gaehna/flow/new2/system/extensions/core.php(685): require('/usr/www/users/...') #4 /usr/www/users/gaehna/flow/new2/system/extensions/core.php(667): YellowPage->includeLayout('blog') #5 /usr/www/users/gaehna/flow/new2/system/extensions/core.php(618): YellowPage->parsePageLayout('blog') #6 /usr/www/users/gaehna/flow/new2/system/extensions/core.php(214): YellowPage->parsePage() #7 /usr/www/users/gaehna/flow/new2/system/extensions/core.php(149): YellowCore->sendPage() # in /usr/www/users/gaehna/flow/new2/system/extensions/comments.php on line 61`

Can you help to solve this problem?
Thanks in advance
René

**Comment by GiovanniSalmeri** [comment]
Hi René!

I'm sorry for the problem, I have to update the plugin to the new
API of Yellow... I hope to do everything tomorrow!

> I added your plugin to my dev stage and it seems that i't was'nt
> installed correctly, as I can't find the files in my extension
> directory after putting the zip file there. So I added the files
> manually and also added the parameters to the yellow-system.ini. But
> when I access a blog post, I always see the following error
> (debugging enabled). How can I solve this?

**Comment by flschr** [comment]
HI Giovanni, any news on this task? :-)

**Comment by GiovanniSalmeri** [comment]
Sorry for the delay, I hope I will publish tomorrow the update...

**Comment by markseuffert** [comment]
Happy to help if there are any problems with the new API. Cheers.

**Comment by GiovanniSalmeri** [comment]
@rrrrfischer Sorry for the delay, now \<ins\>almost\</ins\> all should be in order with the new API. I have not tested thoroughly the extension, please tell me if something does not work as intended. (Thank you @markseu !)

**Comment by flschr** [comment]
Thanks for your work. Unfortunately I didn't get it to work. Comments fields are show, but after submitting a comment I see the "something went wrong" message. No comments get saved. How can I debug what happens to give you more details?

Update: Enabled debug=1 and got this details. Hope you can help? :-)

Notice: Trying to get property 'comment' of non-object in /usr/www/users/gaehna/system/extensions/comments.php on line 207

Fatal error: Uncaught Error: Call to undefined function strreplaceu() in /usr/www/users/gaehna/system/extensions/comments.php:211 Stack trace: #0 /usr/www/users/gaehna/system/extensions/comments.php(303): YellowComments->saveComments(true) #1 /usr/www/users/gaehna/system/extensions/comments.php(66): YellowComments->processSend() #2 /usr/www/users/gaehna/system/extensions/comments.php(141): YellowComments->onParseContentShortcut(Object(YellowPage), 'comments', '', true) #3 /usr/www/users/gaehna/system/extensions/core.php(855): YellowComments->onParsePageExtra(Object(YellowPage), 'comments') #4 /usr/www/users/gaehna/system/layouts/blog.html(18): YellowPage->getExtra('comments') #5 /usr/www/users/gaehna/system/extensions/core.php(685): require('/usr/www/users/...') #6 /usr/www/users/gaehna/system/extensions/core.php(667): YellowPage->includeLayout('blog') #7 /usr/www/users/gaehna/system/extensions/core.php(618): YellowPage->parsePageLayout('blog') #8 /usr/www/users/gaehna/system/extensions/core.php(214): YellowPage->parsePag in /usr/www/users/gaehna/system/extensions/comments.php on line 211

**Comment by GiovanniSalmeri** [comment]
I am really sorry, there were some obsolete functions left! Now I have corrected them, but I have not yet tested the whole extension... Tomorrow I will doublecheck all. Thank for your patience!

**Comment by flschr** [comment]
Just replaced the comments.php to give it a quick try. Comments now will be saved, but publishing is running into a error:

Notice: Undefined property: YellowCore::$text in /usr/www/users/gaehna/system/extensions/comments.php on line 82

Fatal error: Uncaught Error: Call to a member function normaliseDate() on null in /usr/www/users/gaehna/system/extensions/comments.php:82 Stack trace: #0 /usr/www/users/gaehna/system/extensions/comments.php(141): YellowComments->onParseContentShortcut(Object(YellowPage), 'comments', '', true) #1 /usr/www/users/gaehna/system/extensions/core.php(855): YellowComments->onParsePageExtra(Object(YellowPage), 'comments') #2 /usr/www/users/gaehna/system/layouts/blog.html(18): YellowPage->getExtra('comments') #3 /usr/www/users/gaehna/system/extensions/core.php(685): require('/usr/www/users/...') #4 /usr/www/users/gaehna/system/extensions/core.php(667): YellowPage->includeLayout('blog') #5 /usr/www/users/gaehna/system/extensions/core.php(618): YellowPage->parsePageLayout('blog') #6 /usr/www/users/gaehna/system/extensions/core.php(214): YellowPage->parsePage() #7 /usr/www/users/gaehna/system/extensions/core.php(149): YellowCore->sendPage() #8 /usr/www/users/gaehna/system/extensions/core.php(118): YellowCore->processRequest('https', ' in /usr/www/users/gaehna/system/extensions/comments.php on line 82

**Comment by flschr** [comment]
And finally, when I access a page with saved comments I receive this error message:

Notice: Undefined index: published in /usr/www/users/gaehna/system/extensions/comments.php on line 17

Notice: Undefined property: YellowCore::$text in /usr/www/users/gaehna/system/extensions/comments.php on line 80

Fatal error: Uncaught Error: Call to a member function normaliseDate() on null in /usr/www/users/gaehna/system/extensions/comments.php:80 Stack trace: #0 /usr/www/users/gaehna/system/extensions/comments.php(139): YellowComments->onParseContentShortcut(Object(YellowPage), 'comments', '', true) #1 /usr/www/users/gaehna/system/extensions/core.php(855): YellowComments->onParsePageExtra(Object(YellowPage), 'comments') #2 /usr/www/users/gaehna/system/layouts/blog.html(18): YellowPage->getExtra('comments') #3 /usr/www/users/gaehna/system/extensions/core.php(685): require('/usr/www/users/...') #4 /usr/www/users/gaehna/system/extensions/core.php(667): YellowPage->includeLayout('blog') #5 /usr/www/users/gaehna/system/extensions/core.php(618): YellowPage->parsePageLayout('blog') #6 /usr/www/users/gaehna/system/extensions/core.php(214): YellowPage->parsePage() #7 /usr/www/users/gaehna/system/extensions/core.php(149): YellowCore->sendPage() #8 /usr/www/users/gaehna/system/extensions/core.php(118): YellowCore->processRequest('https', ' in /usr/www/users/gaehna/system/extensions/comments.php on line 80

**Comment by GiovanniSalmeri** [comment]
@rrrrfischer Could you try the extension now? There were still some API updates pending, and also a nasty bug that probably no-one stumbled upon before. The whole extension needs some housekeeping and several simplifications, but now it seems to work...

@markseu Is there some advantage in using for example `strlenu` instead of `mb_strlen`? Should I use the former in Yellow extensions?

**Comment by markseuffert** [comment]
@GiovanniSalmeri No, not from a technical point of view. The reason for introducing this function  is that whenever you see `strlen` in a PHP code, it's not clear if the developer meant length in UTF-8 characters or bytes. That's why I recommend to make your intention clear with `strlenu` or `strlenb`. There are [additional string functions in the Yellow API](https://datenstrom.se/yellow/help/api-for-developers#yellow-string).

**Comment by GiovanniSalmeri** [comment]
Big commit with 132 additions and 168 deletions. I refactored the code with several simplifications. One option added for getting rid of the consent checkbox, which to the best of my knowledge is not required by GDPR. Added a dynamic count of characters inserted (a JS one-liner :-). Now the text is formatted with an ad-hoc function with a tiny subset of Markdown. No change in the format of comments. Hope it will be useful!
